src/inter-aut.cc

```diff
@@ -229,13 +229,14 @@ namespace {
                     break;
                 case mata::FormulaNode::Type::OPERATOR:
                     for (int j = static_cast<int>(opstack.size())-1; j >= 0; --j) {
-                        size_t j_size_t{ static_cast<size_t>(j) };
-                        assert(!opstack[j_size_t].is_operand());
-                        if (opstack[j_size_t].is_leftpar())
+                        auto formula_node_opstack_it{ opstack.begin() + j };
```

Comment:
Isn't this just iterating from end to begin? So we could have
`for (auto formula_node_opstack_it = opstack.rbegin(); formula_node_opstack_it != opstack.rend(); --formula_node_opstack_it)`?

Reply:
It is not iterating since the container is a vector which has random access iterator. That means that we can compute offset directly, without iterating over each element in the range from the current iterator to the offset.

Reply:
I think you are right. It should be possible to refactor this as you say. I will investigate.

Reply:
The int arithmetic is there so you can do `erase()` in the vector. It will not work with a reverse iterator.

Reply:
I think it could be done somehow with `erase()`, because it returns iterator, but a forward one.
But moving erasing outside the loop would probably be the best.